tests/components/homeassistant/triggers/test_time.py

```diff
@@ -877,3 +877,218 @@ async def test_if_at_template_limited_template(
     await hass.async_block_till_done()
 
     assert "is not supported in limited templates" in caplog.text
+
+
+async def test_if_fires_using_weekday_single(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+    service_calls: list[ServiceCall],
+) -> None:
+    """Test for firing on a specific weekday."""
+    # Set time to Monday at 5:00:00
+    monday = dt_util.now().replace(hour=5, minute=0, second=0, microsecond=0)
+    # Find the next Monday
```

Comment:
```suggestion
    today = dt_util.now().replace(hour=5, minute=0, second=0, microsecond=0)
```
this is not "monday", its today at 5:00:00, right?